src/third/redux-persist/createPersistor.js

```diff
@@ -37,18 +37,34 @@ export default function createPersistor (store, config) {
   store.subscribe(() => {
     if (paused || writeInProgress) return;
 
-    writeOnce(store.getState());
+    write();
   })
 
+  async function write() {
+    // Take the lock.
+    writeInProgress = true;
+    // Then yield so the `subscribe` callback can promptly return.
+    await new Promise(r => setTimeout(r, 0));
+
+    try {
+      let state = undefined;
+      // eslint-disable-next-line no-cond-assign
+      while ((state = store.getState()) !== lastWrittenState) {
+        await writeOnce(state);
+      }
+    } finally {
+      // Release the lock, so the next `subscribe` will start the loop again.
```

Comment:
nit: next `subscribe` *callback*, really -- `subscribe` itself we call just the once

Reply:
Ha, and I see this line comes straight from my draft:
https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/data.20storage/near/1214393
:shrug: